### DISCUSSION TITLE
怎么设置参考文献中的超链接为蓝色，并调整参考文献的字体大小？

### DISCUSSION THREAD

**Discussion by earthmoon** [whutug/whu-thesis#160] [Q&A] [answered]
你好，
我看文末参考文献的超链接是黑色，可以调成蓝色吗？个人感觉蓝色更友好一点。
参考文献的字体感觉也有点大，怎么把它调小一点呢？
谢谢

**Comment by sikouhjw** [comment, marked as answer]
1. 一般来说超链接不需要为彩色，因为黑白打印时彩色会很淡
2. 参考文献的超链接是用 `\url` 输出的，那建议你去看 `texdoc hyperref` 和模板相关设置
3. 参考文献字体大小的问题，建议你先去看论文规范有没有涉及

**Reply by earthmoon** [reply]
你好，我使用`\usepackage[urlcolor=blue]{hyperref}`设置颜色似乎无效。

**Reply by earthmoon** [reply]
你好，`\DeclareUrlCommand\url{\color{blue}}`这个代码可以修改颜色为蓝色。

**Reply by earthmoon** [reply]
> 你好，`\DeclareUrlCommand\url{\color{blue}}`这个代码可以修改颜色为蓝色。

此方案不行，会使得超链接失效

**Reply by sikouhjw** [reply]
由于
https://github.com/whutug/whu-thesis/blob/6f19c2e0d34bbfc752f86a7ea505154a185ff595/whu-thesis.cls#L726-L727
导致超链接颜色全部隐藏，可以在导言区写下面的代码来获得颜色
```tex
\AtEndPreamble
  {
    \hypersetup
      {
        colorlinks  = true,% 全部超链接都会有颜色
        urlcolor    = blue,% 设置 `\url` 的颜色为蓝色
      }
  }
```

**Reply by earthmoon** [reply]
> 由于
> 
> https://github.com/whutug/whu-thesis/blob/6f19c2e0d34bbfc752f86a7ea505154a185ff595/whu-thesis.cls#L726-L727
> 
> 
> 导致超链接颜色全部隐藏，可以在导言区写下面的代码来获得颜色
> ```
> \AtEndPreamble
>   {
>     \hypersetup
>       {
>         colorlinks  = true,% 全部超链接都会有颜色
>         urlcolor    = blue,% 设置 `\url` 的颜色为蓝色
>       }
>   }
> ```

谢谢，这个方案可以解决。调整文末参考文献的字体大小可以帮忙找一下吗？谢谢！

**Reply by earthmoon** [reply]
这个全部都有颜色看起来有点不合适，有办法只在参考文献显示超链接为蓝色吗？

**Reply by sikouhjw** [reply]
> 这个全部都有颜色看起来有点不合适，有办法只在参考文献显示超链接为蓝色吗？

`texdoc hyperref`，将其他颜色调成黑色。

**Reply by earthmoon** [reply]
> > 这个全部都有颜色看起来有点不合适，有办法只在参考文献显示超链接为蓝色吗？
> 
> `texdoc hyperref`，将其他颜色调成黑色。

谢谢！目前用此代码无问题：

> \AtEndPreamble
>   {
>     \hypersetup
>       {
>         colorlinks  = true,% 全部超链接都会有颜色
>         urlcolor    = blue,% 设置 `\url` 的颜色为蓝色
>         citecolor   = black,%设置正文引用文献颜色为黑色
>         linkcolor   = black,%设置文内跳转连接为黑色
>       }
>   }
>

**Reply by sikouhjw** [reply]
参考文献字体大小是 bug
https://github.com/whutug/whu-thesis/blob/6f19c2e0d34bbfc752f86a7ea505154a185ff595/whu-thesis.cls#L147-L153
调整字体大小的代码没有起效果，临时方法是
```tex
\BeforeBeginEnvironment{document}
  {
    \renewcommand{\bibfont}{\zihao{5}}
  }
```
@tanukihee @xkwxdyy 修一下

**Reply by earthmoon** [reply]
> 参考文献字体大小是 bug
> 
> https://github.com/whutug/whu-thesis/blob/6f19c2e0d34bbfc752f86a7ea505154a185ff595/whu-thesis.cls#L147-L153
> 
> 
> 调整字体大小的代码没有起效果，临时方法是
> ```
> \BeforeBeginEnvironment{document}
>   {
>     \renewcommand{\bibfont}{\zihao{5}}
>   }
> ```
> 

这些代码加到哪呢？

**Reply by earthmoon** [reply]
> ```
> \BeforeBeginEnvironment{document}
>   {
>     \renewcommand{\bibfont}{\zihao{5}}
>   }
> ```

谢谢！加到导言区即可。

**Reply by xkwxdyy** [reply]
> 参考文献字体大小是 bug
> 
> https://github.com/whutug/whu-thesis/blob/6f19c2e0d34bbfc752f86a7ea505154a185ff595/whu-thesis.cls#L147-L153
> 
> 
> 调整字体大小的代码没有起效果，临时方法是
> ```
> \BeforeBeginEnvironment{document}
>   {
>     \renewcommand{\bibfont}{\zihao{5}}
>   }
> ```
> 
> @tanukihee @xkwxdyy 修一下

收到，感谢指出
